File: py3/169.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d = dict()
        for num in nums:
            if d.get(num):
                d[num] += 1
            else:
                d[num] = 1
        
        max_num, max_count = 0, 0
        for num, count in d.items():
            if count > max_count:
                max_num = num
                max_count = count
            if max_count >= len(nums)/2:
                return max_num
        
        return max_num

File: go/src/169.go
func majorityElement(nums []int) int {
	if len(nums) < 1 {
		return -1
	}

	ans := nums[0]
	times := 1
	for i := 1; i < len(nums); i++ {
		if times == 0 {
			ans = nums[i]
			times = 1
		} else if ans == nums[i] {
			times++
		} else {
			times--
		}
	}

	return ans
}